### PR TITLE
exclude generated files from typechecking

### PR DESCRIPTION
**Pull request by tjheffner**
## Description
After creating the symlink to vets-website (`yarn dev`), other commands would fail: name `yarn test:types` and `yarn build` or `yarn export` (both of which check types) due to typescript looking at all of the generated files from vets-website. We don't need to be checking those, they aren't ours. And they cause our stuff to break if we do.

```
yarn test:types
<--- Last few GCs --->

[8923:0x148040000]    15316 ms: Scavenge 4041.1 (4113.6) -> 4040.0 (4117.9) MB, 5.5 / 0.0 ms  (average mu = 0.564, current mu = 0.386) allocation failure;
[8923:0x148040000]    15330 ms: Scavenge 4045.1 (4118.1) -> 4043.6 (4136.1) MB, 7.5 / 0.0 ms  (average mu = 0.564, current mu = 0.386) allocation failure;
[8923:0x148040000]    15806 ms: Mark-sweep 4055.6 (4136.1) -> 4051.4 (4151.1) MB, 460.6 / 0.0 ms  (average mu = 0.428, current mu = 0.162) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0x104e966e4 node::Abort() [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 2: 0x104e968c8 node::ModifyCodeGenerationFromStrings(v8::Local<v8::Context>, v8::Local<v8::Value>, bool) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 3: 0x104fed60c v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 4: 0x105197ec4 v8::internal::EmbedderStackStateScope::EmbedderStackStateScope(v8::internal::Heap*, v8::internal::EmbedderStackStateScope::Origin, cppgc::EmbedderStackState) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 5: 0x10519686c v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 6: 0x10518a900 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 7: 0x10518b144 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 8: 0x1051710c4 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
 9: 0x1055116c0 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
10: 0x10586104c Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_NoBuiltinExit [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
11: 0x10a779904
12: 0x10a7ae51c
13: 0x10a76486c
14: 0x10a7a0494
15: 0x10a75c128
16: 0x10a7b87fc
17: 0x10a7ae6c0
18: 0x10a76486c
19: 0x10a7a0494
20: 0x10a75c128
21: 0x10a762890
22: 0x10a7733d8
23: 0x10a775c7c
24: 0x10a7631e0
25: 0x10a751cc0
26: 0x10a78a770
27: 0x10a774548
28: 0x10a75cb38
29: 0x10a7cfe0c
30: 0x10a77369c
31: 0x10a775c7c
32: 0x10a7631e0
33: 0x10a751cc0
34: 0x10a78a770
35: 0x10a774548
36: 0x10a75cb38
37: 0x10a848130
38: 0x10a76498c
39: 0x10a7a0494
40: 0x10a75c128
41: 0x10a762890
42: 0x10a7733d8
43: 0x10a775c7c
44: 0x10a7631e0
45: 0x10a751cc0
46: 0x10a78a770
47: 0x10a774548
48: 0x10a75cb38
49: 0x10a7cfe0c
50: 0x10a77369c
51: 0x10a775c7c
52: 0x10a7631e0
53: 0x10a751cc0
54: 0x10a78a770
55: 0x10a774548
56: 0x10a75cb38
57: 0x10a848130
58: 0x10a76498c
59: 0x10a7a0494
60: 0x10a75c128
61: 0x10a762890
62: 0x10a7733d8
63: 0x10a775c7c
64: 0x10a7631e0
65: 0x10a751cc0
66: 0x10a78a770
67: 0x10a774548
68: 0x10a75cb38
69: 0x10a7cfe0c
70: 0x10a77369c
71: 0x10a775c7c
72: 0x10a7631e0
73: 0x10a751cc0
74: 0x10a78a770
75: 0x10a774548
76: 0x10a7b85b8
77: 0x10a72d35c
78: 0x10a7773fc
79: 0x10a8a3f88
80: 0x10a8121f0
81: 0x10a773644
82: 0x10a775c7c
83: 0x10a7631e0
84: 0x10a751cc0
85: 0x10a78a770
86: 0x10a774548
87: 0x10a75cb38
88: 0x10a7cfe0c
89: 0x10a77369c
90: 0x10a775c7c
91: 0x10a7631e0
92: 0x10a751cc0
93: 0x10a78a770
94: 0x10a774548
95: 0x10a80f974
96: 0x10a7ba65c
97: 0x10a8a3f88
98: 0x10a80dae0
99: 0x10a750584
100: 0x10a764a18
101: 0x10a7a0494
102: 0x10a75c128
103: 0x10a762890
104: 0x10a7733d8
105: 0x10a775c7c
106: 0x10a7631e0
107: 0x10a751cc0
108: 0x10a78a770
109: 0x10a774548
110: 0x10a75cb38
111: 0x10a7cfe0c
112: 0x10a77369c
113: 0x10a775c7c
114: 0x10a7631e0
115: 0x10a751cc0
116: 0x10a78a770
117: 0x10a774548
118: 0x10a75cb38
119: 0x10a848130
120: 0x10a76498c
121: 0x10a7a0494
122: 0x10a3b337c
123: 0x10a817cbc
124: 0x10a3addc4
125: 0x10a3b0c68
126: 0x10a40a128
127: 0x10a86b760
128: 0x10a3b618c
129: 0x10a34bf84
130: 0x10a861098
131: 0x10a34c310
132: 0x10a890e70
133: 0x10a70d1c4
134: 0x10a35b894
135: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
136: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
137: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
138: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
139: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
140: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
141: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
142: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
143: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
144: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
145: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
146: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
147: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
148: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
149: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
150: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
151: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
152: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
153: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
154: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
155: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
156: 0x1057ec198 Builtins_InterpreterEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
157: 0x1057ea4d0 Builtins_JSEntryTrampoline [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
158: 0x1057ea164 Builtins_JSEntry [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
159: 0x105119d40 v8::internal::(anonymous namespace)::Invoke(v8::internal::Isolate*, v8::internal::(anonymous namespace)::InvokeParams const&) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
160: 0x105119280 v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
161: 0x1050094ac v8::Function::Call(v8::Local<v8::Context>, v8::Local<v8::Value>, int, v8::Local<v8::Value>*) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
162: 0x104f02d64 node::Realm::ExecuteBootstrapper(char const*, std::__1::vector<v8::Local<v8::Value>, std::__1::allocator<v8::Local<v8::Value>>>*) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
163: 0x104e606d0 node::StartExecution(node::Environment*, char const*) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
164: 0x104e60630 node::StartExecution(node::Environment*, std::__1::function<v8::MaybeLocal<v8::Value> (node::StartExecutionCallbackInfo const&)>) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
165: 0x104de0708 node::LoadEnvironment(node::Environment*, std::__1::function<v8::MaybeLocal<v8::Value> (node::StartExecutionCallbackInfo const&)>) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
166: 0x104ed32a4 node::NodeMainInstance::Run() [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
167: 0x104e63338 node::LoadSnapshotDataAndRun(node::SnapshotData const**, node::InitializationResult const*) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
168: 0x104e635f0 node::Start(int, char**) [/Users/heffner/.nvm/versions/node/v18.17.1/bin/node]
169: 0x197cf3f28 start [/usr/lib/dyld]
```

I also added a small placeholder meta tag to fix an error thrown on individual news story pages. 
I also added a small fetch function for one more missing data file that was being logged in the console. That will probably need to be adjusted as part of the environment check too.

## Testing done
I was able to run `yarn test:types` and `yarn build` and `yarn export` without it throwing up on the typechecking step.

## QA steps
```[tasklist]
- [x] Automated tests have passed
- [x] Tugboat environment was generated without errors
- [ ] Run `yarn dev` locally to set up the symlink to the vets-website repo. Follow the directions in the readme if you do not have it set up in an adjacent folder already.
- [ ] Run `yarn build` and verify that it makes it past the lint & typechecking portion of the script.
- [ ] Navigate to an individual news story page and verify no errors are thrown.
- [ ] No message in the browser console log about `/public/data/vamc-ehr.json`
```

### CODE REVIEW

File: src/pages/[[...slug]].tsx [hunk 1]
```diff
@@ -29,6 +29,8 @@ export default function ResourcePage({ resource, globalElements }) {
     <Wrapper bannerData={globalElements.bannerData}>
       <Head>
         <title>{title}</title>
+        {/* todo: do all meta tags correctly, this fixes an error on news story */}
+        <meta property="og:url" content={path} />
```
Comment (ryguyk) [inline src/pages/[[...slug]].tsx:33]: **MUST (or at least probably should eventually)**
I'm not sure this is going to work. `path` is not the current url in all cases. For listing pages, `path` is the url of the first page.

We can easily get what we need, but we'll need to make a few changes (including maybe naming, since it does feel like `path` is the current url) in `getStaticProps`.

**Review by ryguyk — DISMISSED** [review]
A question and a couple change requests that probably should happen now or eventually. Marking as requesting changes, but open to kicking to another ticket if needed.

File: scripts/setup-dev.js [hunk 2]
```diff
@@ -21,4 +22,33 @@ const symlinkPath = path.resolve(__dirname, '..', 'public', 'generated')
...
+              // eslint-disable-next-line no-console
+              console.log('vamc-ehr data fetched successfully!')
+            })
+            .catch((err) => {
+              console.error('Error with cms data directory: ', err)
+            })
+        })
+        .catch((err) => {
+          console.error('Error fetching cms data from va.gov: ', err)
+        })
```
[the remark this answers is not in the document]
Comment (tjheffner) [reply scripts/setup-dev.js:46]: yeah, this is a good callout. not sure what i was thinking here. will update

File: .eslintignore [hunk 3]
```diff
@@ -1 +1,8 @@
 src/**/__tests__/*
+public/generated/*
+/node_modules/*
+/coverage/*
+/.swc/*
+/.next/*
+/out/*
+/storybook-static/*
```
[the remark this answers is not in the document]
Comment (tjheffner) [reply .eslintignore:8]: A lot of the files in these directories are html or json, but it was still scanning through them because of our very permissive globs in tsconfig.json (eslint reads from that)

```
  "include": [
    "next-env.d.ts",
    "additional.d.ts",
    "src/**/*",
    "**/*.ts",
    "**/*.tsx",
    "**/*.js",
    "**/*.jsx"
  ],
```

File: scripts/setup-dev.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ const path = require('path')
 
 const target = path.resolve('../vets-website/build/localhost/generated')
 const symlinkPath = path.resolve(__dirname, '..', 'public', 'generated')
+const cmsDataPath = path.resolve(__dirname, '..', 'public', 'data', 'cms')
```
[the remark this answers is not in the document]
Comment (tjheffner) [reply scripts/setup-dev.js:14]: updated and added a comment

File: src/pages/[[...slug]].tsx (hunk 1 above)
Comment (tjheffner) [reply src/pages/[[...slug]].tsx]: Heard chef. I've removed this for now with a hardcoded 'foo'. We will handle meta info properly in a follow up ticket.

File: scripts/setup-dev.js [hunk 5]
```diff
@@ -2,7 +2,14 @@
...
+// No __dirname here because we want it to reach the adjacent folder outside of next-build
+const target = path.resolve(
+  '..',
+  'vets-website',
+  'build',
+  'localhost',
+  'generated'
+)
 const symlinkPath = path.resolve(__dirname, '..', 'public', 'generated')
 const cmsDataPath = path.resolve(__dirname, '..', 'public', 'data', 'cms')
```
Comment (ryguyk) [inline scripts/setup-dev.js:14]: Thanks for this change and, especially, the comment attempting to explain what's going on. I'm happy to move forward on this, but I am adding a bit more commentary below in an attempt to clarify my initial line of questioning.

_WARNING: Theoretical discussion. It's certainly ok to just move on with life and not read this :)_

I immediately see how `target` is different from the other two. It's a directory outside `next-build` whereas the other two are directories within `next-build`. So it's not immediately crazy to me to see two different treatments divided by this line of separation. But I'm still not sure it makes sense to me. I'm not sure that difference should determine the use (or lack of use) of `__dirname`.

To me, the question of the usage of `__dirname` as it relates to path resolution boils down to whether it makes more sense to provide "directions" about the "address" of a particular resource relative to a certain directory, or relative to the root. If we use `__dirname`, we are telling the thing that needs those directions to the address that it makes most sense to start from the directory of the script being executed and go from there. This makes a ton of sense when the resource is a subdirectory of the directory of the current script because the relationship will never be independent. If the directory of the current script got moved down a few levels, things would still work. In general, this makes a lot of sense when the physical relationship (in the directory structure) _between_ the two resources is constant.

On the other hand, giving directions from the root makes sense when the resource in question isn't necessarily related to the current directory. In this case, I think it's arguable that this case is the more accurate. We know that `public` should always be a direct child of the root of the project. We cannot say the same about the location of the directory that houses `setup-dev.js`. To demonstrate, let's consider what would happen if we decided to add more scripts, and so we introduced another directory level:
```
- /public
  - /generated 
- /scripts
  - /dev
    - setup-dev.js
    - some-other-dev-script.js
  - /staging
    - setup-staging.js
  - /prod
    - setup-prod.js 
``` 
In this case, the code would have to be changed to accommodate. Sure, if we decided to move `public` to a different level, we'd have to change the code, but we'd have to change the code _in both cases_ (using `__dirname` or not).

**TL;DR** All three of these directory locations feel like they are best described by their directory location relative to the root of `next-build`, rather than to the directory of the current script.

File: scripts/setup-dev.js [hunk 6]
```diff
@@ -28,22 +35,23 @@ const cmsDataPath = path.resolve(__dirname, '..', 'public', 'data', 'cms')
...
+
+      await fs
+        .mkdirp(cmsDataPath)
+        .catch((err) =>
+          console.error('Error creating cms data directory: ', err)
+        )
+
+      fs.writeJson(`${cmsDataPath}/vamc-ehr.json`, data).catch((err) =>
+        console.error('Error writing data file: ', err)
+      )
```
Comment (ryguyk) [inline scripts/setup-dev.js:54]: This looks better (and, again, will work), but, at least as I understand things, it's still not idiomatic/best practice. Generally, `await` is used within `try...catch` blocks in order to function more like synchronous code. This is still mixing that syntax with the more traditional `Promise` handling that uses chaining.

Two possible fixes:
1. Remove `await` in the first iteration and move on. That is probably the easiest fix, but probably not the best. We are using `await` fairly consistently in this project, and I think this code is best implemented to match that convention.
2. 
```
let data
try {
  data = await fetch('https://va.gov/data/cms/vamc-ehr.json')
} catch(err) {
  console.error('Error fetching cms data from va.gov: ', err)
}

// eslint-disable-next-line no-console
console.log('vamc-ehr data fetched successfully!')

try {
  await fs.mkdirp(cmsDataPath)
} catch(err) {
  console.error('Error creating cms data directory: ', err)
}

try {
  await fs.writeJson(`${cmsDataPath}/vamc-ehr.json`, data)
} catch(err) {
  console.error('Error writing data file: ', err)
)
```

As I type that, though, I recognize that the calls are dependent on one another, and those dependencies seem to be covered appropriately in the `Promise` chain approach. The endless indentation is hard on the eyes, and one big benefit of this approach using `await`, but we'd need to exit appropriately if calls fail.

File: scripts/setup-dev.js (hunk 5 above)
Comment (tjheffner) [reply scripts/setup-dev.js]: this makes a lot of sense, i've added __dirname to target as well. thank you.

**Review by jtmst — APPROVED** [review]

File: .eslintrc.json [hunk 7]
```diff
@@ -46,7 +46,7 @@
       "error",
       {
         "devDependencies": [
-          "**/*.test.*",
+          "src/**/*.test.*",
```
Comment (tjheffner) [inline .eslintrc.json:49]: this overly greedy glob was causing OOM errors for me.

File: tsconfig.json [hunk 8]
```diff
@@ -41,12 +41,10 @@
     "next-env.d.ts",
     "additional.d.ts",
     "src/**/*",
-    "**/*.ts",
-    "**/*.tsx",
-    "**/*.js",
-    "**/*.jsx"
+    "*/*.{ts|tsx|js|jsx}",
```
Comment (tjheffner) [inline tsconfig.json:44]: the `**/*` globs here and in .eslintrc.json were causing the OOM for me. restricting them slightly, eslint travels through the files without hanging now.
